app/src/ui/history/commit-list.tsx

```diff
@@ -143,10 +151,20 @@ export class CommitList extends React.Component<ICommitListProps, {}> {
             commits: this.props.commitSHAs,
             gitHubUsers: this.props.gitHubUsers,
             localCommitSHAs: this.props.localCommitSHAs,
+            commitLookupHash: this.commitLookupHash([
+              ...this.props.commitLookup.values(),
+            ]),
```

Comment:
Can we use the commits stored in `this.props.commitSHAs` instead of the ones in `this.props.commitLookup` to invalidate the commit list?

I'm asking because `commitLookup` will contain all the commits that have been loaded in the repository, while `commitSHAs` will have the ones that get actually rendered by the list, so it may be less work.

The only downside is that we'll need to do a lookup for each of them (but that should be really fast since the lookup object is a `Map`)